### ISSUE TITLE
[ios] [flutter] GeneratedPluginRegistrant.m Module not found. 

### ISSUE THREAD

**Issue by rushabhshah065** [flutter/flutter#53573]
Any pub package which I have used in my project generates the same error.
**XCode Version 11.4**
**COCOAPODS: 1.9.1**

<img width="1079" alt="Screen Shot 2020-03-30 at 6 36 03 PM" src="https://user-images.githubusercontent.com/48256486/77915824-bfc18800-72b5-11ea-8141-800518e63ced.png">

**Comment by VladyslavBondarenko** [comment]
Hi @rushabhshah065 
I came across the your comment https://github.com/flutter/flutter/issues/43986#issuecomment-605987016.
Are you running from `Runner.xcworkspace`, right?

**Comment by rushabhshah065** [comment]
> Hi @rushabhshah065
> I came across the your comment [#43986 (comment)](https://github.com/flutter/flutter/issues/43986#issuecomment-605987016).
> Are you running from `Runner.xcworkspace`, right?

Yes.

**Comment by VladyslavBondarenko** [comment]
@rushabhshah065
could you provide you `flutter doctor -v` and `flutter run --verbose`?

**Comment by ryan-sf** [comment]
I can verify that this occurs for me personally when running `flutter build ios --no-codesign` with the `flutter_web_auth` plugin.

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.4 19E266,
    locale en-US)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.43.2)
[✓] Connected device (2 available)
```

I get the same behavior if I switch to the beta channel as well.

**Comment by rushabhshah065** [comment]
> @rushabhshah065
> could you provide you `flutter doctor -v` and `flutter run --verbose`?

This happens when I am trying to add a firebase database dependency on my project. Until that, it's work fine, but after adding it generates errors as I mentioned above.

**Comment by intellitour** [comment]
Same issue.
<img width="2032" alt="image" src="https://user-images.githubusercontent.com/5827270/79130858-061aea80-7d7e-11ea-940c-e19828570055.png">

<details>
<summary>Flutter Doctor output</summary>
```shell
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.4 19E266, locale pt-BR)
    • Flutter version 1.12.13+hotfix.9 at /Users/pedro/dev/Flutter/sdk
    • Framework revision f139b11009 (2 weeks ago), 2020-03-30 13:57:30 -0700
    • Engine revision af51afceb8
    • Dart version 2.7.2

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/pedro/dev/Android/sdk/
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/pedro/dev/Android/sdk/
    • Java binary at: /Users/pedro/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/192.6308749/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.9.1

[✓] Android Studio (version 3.6)
    • Android Studio at /Users/pedro/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/192.6308749/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[!] IntelliJ IDEA Ultimate Edition (version 2020.1)
    • IntelliJ at /Users/pedro/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.43.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.
```
</details>

<details>
<summary>iOS Build error</summary>
```shell
Building br.med.maisvida.plataformaBeneficiarioFlutter for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: S75524PW58
Running pod install...                                              2,6s
Running Xcode build...                                          
Xcode build done.                                           49,3s
Failed to build iOS app
Error output from Xcode build:
↳
    2020-04-13 12:00:25.410 xcodebuild[22249:152542] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7f9b850cb120), iPhone de Pedro Henrique, iPhone, 13.4 (17E255), 00008020-0010696814C2002E> == Underlying device preparation errors ==
    2020-04-13 12:00:25.410 xcodebuild[22249:152542] [MT] iPhoneConnect: Failed _shouldMakeReadyForDevelopment check even though device is not locked by passcode.
    Domain: com.apple.platform.iphoneos
    Code: 5
    Failure Reason: allowsSecureServices: 1. isConnected: 0. Platform: <DVTPlatform:0x7f9b80732e40:'com.apple.platform.iphoneos':<DVTFilePath:0x7f9b80732bc0:'/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform'>>. DTDKDeviceIdentifierIsIDID: 0
    User Info: {
        DVTDeviceDescription = "\Ud83d\Udcf1<DVTiOSDevice (0x7f9b850cb120), iPhone de Pedro Henrique, iPhone, 13.4 (17E255), 00008020-0010696814C2002E>";
    }
    --
    2020-04-13 12:00:25.410 xcodebuild[22249:152542] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7f9b850cb120), iPhone de Pedro Henrique, iPhone, 13.4 (17E255), 00008020-0010696814C2002E> == END: Underlying device preparation errors ==
    ** BUILD FAILED **


Xcode's output:
↳
    Building AOT snapshot in release mode (ios-release)...          
    Building App.framework for arm64...

    Building App.framework for armv7...
    Building AOT snapshot in release mode (ios-release)...             38,0s
    Built to build/aot/.
    warning: parsing line table prologue at offset 0x6f697463 found unsupported version 0x00
    warning: line table parameters mismatch. Cannot emit.
    note: while processing /Users/pedro/dev/MaisVida/Plataforma/plataforma_beneficiario_flutter/build/aot/armv7/snapshot_assembly.o
    Project /Users/pedro/dev/MaisVida/Plataforma/plataforma_beneficiario_flutter built and packaged successfully.
    /Users/pedro/dev/MaisVida/Plataforma/plataforma_beneficiario_flutter/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'catcher' not found
    @import catcher;
     ~~~~~~~^~~~~~~
    1 error generated.
    /Users/pedro/dev/MaisVida/Plataforma/plataforma_beneficiario_flutter/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'catcher' not found
    @import catcher;
     ~~~~~~~^~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description


```
</details>

**Comment by MisterJimson** [comment]
I'm running into this when running just `flutter run`, not using xcode.

**Comment by liyang1518** [comment]
i hava the issue.
run flutter build ios suscess,but in xcode is not work,the issue is like above.

zhihengxindeMac-mini:fsp_serve_app ies$ flutter doctor -v
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.4 19E287, locale zh-Hans-CN)
    • Flutter version 1.12.13+hotfix.9 at /Users/ies/android/flutter_12_9/flutter
    • Framework revision f139b11009 (6 weeks ago), 2020-03-30 13:57:30 -0700
    • Engine revision af51afceb8
    • Dart version 2.7.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/ies/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 29.0.3
    • ANDROID_HOME = /Users/ies/Library/Android/sdk
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk1.8.0_251.jdk/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_251-b08)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/setup/#android-setup for detailed instructions).

[✓] Connected device (1 available)
    • iPhone X • 40D27C26-3CEE-4466-A0B3-163FF014524B • ios • com.apple.CoreSimulator.SimRuntime.iOS-12-4 (simulator)

! Doctor found issues in 1 category.

![image](https://user-images.githubusercontent.com/47839081/82027774-565fd200-96c7-11ea-88c0-663c725a411a.png)


relate issue [43986](https://github.com/flutter/flutter/issues/43986) i have tried all, but is not work.

![image](https://user-images.githubusercontent.com/47839081/82027389-c4f06000-96c6-11ea-96b9-4e282132f687.png)

![image](https://user-images.githubusercontent.com/47839081/82027559-0123c080-96c7-11ea-9cc3-4949d768eb8b.png)

Is there anything wrong with **Pods_Runner.framework**？

**Comment by GY22** [comment]
having the same issue with the WebView_flutter plugin. is there a fix/solution to this big problem???

**Comment by caiogranero** [comment]
Having this issue with flutter_auth0 plugin too. Any solution for this?

**Comment by passsy** [comment]
@caiogranero Had the same problem when I forked `flutter_auth0`. The iOS pluginClass is missing in the `pubspec.yaml`. You can cherry-pick https://github.com/bits-cr/flutter-auth0-lib/commit/2e03f0b799a2d202c3a922a11f5062ffa9046d7a to solve the problem

**Comment by tanmaypandharipande** [comment]
> Same issue.
> <img alt="image" width="2000" src="https://user-images.githubusercontent.com/5827270/79130858-061aea80-7d7e-11ea-940c-e19828570055.png">
> 
> Flutter Doctor output
> iOS Build error
> ```shell
> Building br.med.maisvida.plataformaBeneficiarioFlutter for device (ios-release)...
> Automatically signing iOS for device deployment using specified development team in Xcode project: S75524PW58
> Running pod install...                                              2,6s
> Running Xcode build...                                          
> Xcode build done.                                           49,3s
> Failed to build iOS app
> Error output from Xcode build:
> ↳
>     2020-04-13 12:00:25.410 xcodebuild[22249:152542] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7f9b850cb120), iPhone de Pedro Henrique, iPhone, 13.4 (17E255), 00008020-0010696814C2002E> == Underlying device preparation errors ==
>     2020-04-13 12:00:25.410 xcodebuild[22249:152542] [MT] iPhoneConnect: Failed _shouldMakeReadyForDevelopment check even though device is not locked by passcode.
>     Domain: com.apple.platform.iphoneos
>     Code: 5
>     Failure Reason: allowsSecureServices: 1. isConnected: 0. Platform: <DVTPlatform:0x7f9b80732e40:'com.apple.platform.iphoneos':<DVTFilePath:0x7f9b80732bc0:'/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform'>>. DTDKDeviceIdentifierIsIDID: 0
>     User Info: {
>         DVTDeviceDescription = "\Ud83d\Udcf1<DVTiOSDevice (0x7f9b850cb120), iPhone de Pedro Henrique, iPhone, 13.4 (17E255), 00008020-0010696814C2002E>";
>     }
>     --
>     2020-04-13 12:00:25.410 xcodebuild[22249:152542] [MT] iPhoneConnect: 📱<DVTiOSDevice (0x7f9b850cb120), iPhone de Pedro Henrique, iPhone, 13.4 (17E255), 00008020-0010696814C2002E> == END: Underlying device preparation errors ==
>     ** BUILD FAILED **
> 
> 
> Xcode's output:
> ↳
>     Building AOT snapshot in release mode (ios-release)...          
>     Building App.framework for arm64...
> 
>     Building App.framework for armv7...
>     Building AOT snapshot in release mode (ios-release)...             38,0s
>     Built to build/aot/.
>     warning: parsing line table prologue at offset 0x6f697463 found unsupported version 0x00
>     warning: line table parameters mismatch. Cannot emit.
>     note: while processing /Users/pedro/dev/MaisVida/Plataforma/plataforma_beneficiario_flutter/build/aot/armv7/snapshot_assembly.o
>     Project /Users/pedro/dev/MaisVida/Plataforma/plataforma_beneficiario_flutter built and packaged successfully.
>     /Users/pedro/dev/MaisVida/Plataforma/plataforma_beneficiario_flutter/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'catcher' not found
>     @import catcher;
>      ~~~~~~~^~~~~~~
>     1 error generated.
>     /Users/pedro/dev/MaisVida/Plataforma/plataforma_beneficiario_flutter/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'catcher' not found
>     @import catcher;
>      ~~~~~~~^~~~~~~
>     1 error generated.
>     note: Using new build system
>     note: Building targets in parallel
>     note: Planning build
>     note: Constructing build description
> ```

```
pod deintegrate
pod install
```
This worked in my case. 👍

**Comment by khausam** [comment]
In my case, the issue was that I had manually edited the podfile to add in my own (non-flutter) pod to the project. However, the flutter build would make no attempt to edit (or overwrite) the podfile, (maybe this is good) so the podfile would not have the changes necessary to support webview_flutter. I figured this all out by creating a brand new flutter project and comparing the new project's generated podfile against my current version. They looked significantly different.

**Comment by ifrontend-xyz** [comment]
https://github.com/flutter/flutter/issues/21989#issuecomment-513286777
This worked in my case.

**Comment by ionda88** [comment]
Same issue here.
In my case the module that caused the error was 'device_info', when i removed it the issue came from other modules. Seems that the issue is not comming from the modules theirselves.
I already tried:

- Updating my pods.
- Removing some modules.
- Removing PodFile and Podfile.lock and re-generating them.

Nothing worked 😞

**Comment by 926** [comment]
Had the same issue,
Little success with commands such as 
$ flutter clean  
$ pod cache clean --all.
Then I upgraded cocoapods on the latest version with 
$ gem install cocoapods --pre

Might have done other stuff, late night, but fixed now

**Comment by softronaut** [comment]
I have same issue and can't release or use TestFlight for my companies app. I'm just stucked...

**Comment by bukenov** [comment]
https://stackoverflow.com/a/59849303

> $sudo gem uninstall cocoapods

> $sudo gem uninstall cocoapods

> $sudo gem install cocoapods -v 1.7.5

The issue was cocoapods version 1.8+.

You also do not need to add anything to do your podfile if using flutter. Adding the dependencies in pubspec.yaml will automatically set up your podfile for you

This worked in my case!!!

**Comment by chachaxw** [comment]
`[✓] Flutter (Channel stable, 1.20.2, on macOS 11.0 20A5374i, locale
    en-CN)
    • Flutter version 1.20.2 at /Users/june/flutter
    • Framework revision bbfbf1770c (6 weeks ago), 2020-08-13 08:33:09
      -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

 
[!] Android toolchain - develop for Android devices (Android SDK 
    version 28.0.3)
    • Android SDK at /Users/june/Library/Android/sdk
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = /Users/june/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    ! Some Android licenses not accepted.  To resolve this, run:
      flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

[✓] Connected device (2 available)
    • Chacha‘s iPhone 7 (mobile) •
      2b3edf4e355693c03e574c61f66dec6b32980644 • ios • iOS 14.0
    • iPhone 11 Pro Max (mobile) •
      50C00EA6-0AA2-4626-B455-A610A8645CBA     • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)

! Doctor found issues in 1 category.`

**Comment by krumpli84** [comment]
Any updated on this? I'm getting headaches, brain damage and ulcers trying to figure this one out. I can't build the project and run on a physical device but this very error shows up when trying to run on a simulator.

**Comment by syrakozz** [comment]
i tried : 
pod install 
it works

**Comment by albert0m** [comment]
after spending hours after this, I noticed that the iOS Deployment Target for the Runner project was set to a default `8.0`. I increased it as it was set to the Pods project and everything works now.

**Comment by mizzb** [comment]
For me the issue happens if run in release mode in Xcode.
tried everything i can.. Pod install, changing cocoa version, changing target version to 9, changed flutter brach to stable.. still the same module not found error. 
Flutter: 1.20.4
Xcode: 12.0.4

**Comment by tania628** [comment]
> Any updated on this? I'm getting headaches, brain damage and ulcers trying to figure this one out. I can build the project and run on a physical device but this very error shows up when trying to run on a simulator.

I'm in the same issue. but I'm glad to known that run on a physical device. thank you.

**Comment by coolnalu** [comment]
Same issue as @mizzb and @tania628. Builds for device not for simulators.

**Comment by RaphaelRevivor** [comment]
What I did was basically open a terminal and go to the my_project/ folder. Then do:
1. flutter clean
2. fluter build ios
(I might have also deleted Podfile, Podfile.lock as well as Pods/ under my_project/ios/ before step 1.)
So if flutter build ios doesn't work, then it won't in xcode either.

**Comment by rj76** [comment]
Any updates on this? None of the above worked for me....

**Comment by tania628** [comment]
After all, no longer works on physical devices either.
So, I removed entire `project/ios` directory and run `flutter create .`, then all settings was redo. AppIcon, LaunchImage, Podfile, setting for plugins...etc.
afterwards steps:
 1. Just in case, `pod setup`, `flutter clean`
 2. Open project in xcode. (Runner.xcworkspace)
 3. Settings such as `Signing & Capabilities`
 4. `flutter build ios`
 5. Run in simulator... Another error!
> Failed to start launchd_sim: could not bind to session, launchd_sim may have crashed or quit responding Domain: com.apple.SimLaunchHostService.RequestError
[launchd_sim crashing: could not create temporary state directory](https://stackoverflow.com/questions/40238266/launchd-sim-crashing-could-not-create-temporary-state-directory)
 6. Run in simulator again... no error and worked!!

I hope this will be helpful for someone.

**Comment by rj76** [comment]
Removing the ios/Podfile worked for me!

**Comment by krumpli84** [comment]
> Removing the ios/Podfile worked for me!
> 
> Flutter generates the ios/Podfile based on dependencies, but if the file already exists, it doesn't do anything, so `pod install` also doesn't do anything, so the plugins won't be found.
If i do that i get "no podfile found..." I have to do pod install, which means i'm back to square one

**Comment by reju1021** [comment]
After I've also tried every solution above none worked. I got (only archiving, trying to run it with Xcode in release mode still throws module not found errors) working by:
1. Update Xcode to current stable release (12.0.1)
2. Remove old Xcode version
3. Delete iOS folder in flutter project
4. flutter create iOS
5. open iOS folder in finder -> double click on workspace
6. select developing team and rename bundle identifier to something unique.
7. flutter build iOS

archiving should work now in xcode

**Comment by matthias-baeuerle** [comment]
I had a similar issue with the following compiler output:

```
/Users/me/flutter_project/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'barcode_scan' not found
    @import barcode_scan;
     ~~~~~~~^~~~~~~~~~~~
```

My project contains multiple build environments with Flutter flavors as described here: https://medium.com/@thomasgallinari/e7a0c973f54e

What solved the issue for me is changing the casing of the build configurations and the corresponding files from values like `Debug-stage` to `Debug-Stage` as the latter is expected by Flutter.

**Comment by krumpli84** [comment]
After weeks of frustration, i found a way to get it to work:
https://github.com/flutter/flutter/issues/66104#issuecomment-706235825

**Comment by PidoCB** [comment]
> After all, no longer works on physical devices either.
> So, I removed entire `project/ios` directory and run `flutter create .`, then all settings was redo. AppIcon, LaunchImage, Podfile, setting for plugins...etc.
> afterwards steps:
> 
> 1. Just in case, `pod setup`, `flutter clean`
> 2. Open project in xcode. (Runner.xcworkspace)
> 3. Settings such as `Signing & Capabilities`

worked for me, tnx..

**Comment by JonathanSchndr** [comment]
I think the problem is, that you open the Runner.xcodeproj file and not the Runner.xcworkspace file.

flutter clean
open Runner.xcworkspace (!!!)
xcode > clean build folder
xcode > product > scheme > runner
xcode > product > destination > any ios device

and proof, if the iOS Version the same MinimumOSVersion (AppFrameworkInfo)
(https://flutter.dev/docs/deployment/ios)

**Comment by sarvaiyajai** [comment]
I have the same issue connectivity module not found tried all solution. @VladyslavBondarenko

**Comment by Milos5611** [comment]
Super weird issue. 
So I'm able to run it locally, to build, to archive in Xcode...
The only thing I can't make it to work is an azure pipeline.  

```
❌  /Users/runner/work/1/s/ios/Runner/GeneratedPluginRegistrant.m:10:9: module 'flutter_secure_storage' not found

@import flutter_secure_storage;
 ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

And I'm hanging here for so long.

**Comment by griotdonation** [comment]
I had the following output error:  fatal error: module 'square_in_app_payments' not found @import square_in_app_payments;

To use Square in app payment, there IOS platform should be set 11.0

![image](https://user-images.githubusercontent.com/55745803/102372802-92c11c80-3f8d-11eb-97a5-bdaf9b8ce34a.png)

iOS Deployment Target for the Runner project was set to a default 9.0. I updated the the deployment target from 9.0 to 11.0 and now i am able to successfully run 'flutter build ios'

![image](https://user-images.githubusercontent.com/55745803/102373059-d025aa00-3f8d-11eb-9323-c6d11e90ffac.png)

**Comment by itwondersteam** [comment]
guys, have tried all suggested but none is really working. Finally got this work by compile it in Xcode. 
For those still having this issue, you can try

```
# let's just try to clean everything first
flutter clean
cd ios/
pod deintegrate
pod cache clean --all

# do the build
flutter build ios

# then you will see the error
ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'file_picker' not
    found
    @import file_picker;
     ~~~~~~~^~~~~~~~~~~
    1 error generated.

# open the Xcode workspace
open ios/Runner.xcworkspace

# do the compilation over there, once success, you can run your `flutter build` again, this time it will work
```

Build Environment
```
gem which cocoapods
/usr/local/lib/ruby/gems/2.7.0/gems/cocoapods-1.10.0/lib/cocoapods.rb

flutter doctor -v
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.5 19F101 darwin-x64, locale en-GB)
```

**Comment by danielmintz** [comment]
I have tried everything on this post (thanks all) and nothing works, unfortunately. The problem started once I upgraded to Firebase_Core and followed the guidelines here [https://firebase.flutter.dev/docs/migration/#1-adding-firebase_core](https://firebase.flutter.dev/docs/migration/#1-adding-firebase_core). I'm at the stage where I am thinking of creating a new flutter project and then transferring files over yet retaining the same identifiers etc for the App Stores.. Maybe that would work. The thought fills with me dread mind in case I mess up especially as I have no problems with the Android App.. . Anyone tried that and know of any good guides to follow?

**Comment by doomkin** [comment]
In `ios/Runner.xcodeproj/project.bpxproj` replace `IPHONEOS_DEPLOYMENT_TARGET = 8.0` to `IPHONEOS_DEPLOYMENT_TARGET = 12.0` in three lines.

**Comment by yaashwardhan** [comment]
The following should work:
1) Open terminal, navigate to project then `flutter clean`
2) then navigate to /ios folder and `pods install`

**Comment by ariana-sozvayc** [comment]
> The following should work:
> 
> 1. Open terminal, navigate to project then `flutter clean`
> 2. then navigate to /ios folder and `pods install`

This worked thanks! @yaashwardhan , took me a while to figure out the cd command

**Comment by ljg6861** [comment]
> I had the following output error: fatal error: module 'square_in_app_payments' not found @import square_in_app_payments;
> 
> To use Square in app payment, there IOS platform should be set 11.0 in the Podfile.
> 
> ![image](https://user-images.githubusercontent.com/55745803/102372802-92c11c80-3f8d-11eb-97a5-bdaf9b8ce34a.png)
> 
> iOS Deployment Target for the Runner project was set to a default 9.0. I updated the the deployment target from 9.0 to 11.0 and now i am able to successfully run 'flutter build ios'
> 
> ![image](https://user-images.githubusercontent.com/55745803/102373059-d025aa00-3f8d-11eb-9323-c6d11e90ffac.png)

This fixed it for me. 
Note: In my podfile it was set to 12.0, but it was NOT set there in Xcode. You need to set the number in Xcode as well as the podfile.

**Comment by debashreejoshi** [comment]
> After all, no longer works on physical devices either.
> So, I removed entire `project/ios` directory and run `flutter create .`, then all settings was redo. AppIcon, LaunchImage, Podfile, setting for plugins...etc.
> afterwards steps:
> 
> 1. Just in case, `pod setup`, `flutter clean`
> 2. Open project in xcode. (Runner.xcworkspace)
> 3. Settings such as `Signing & Capabilities`
> 4. `flutter build ios`
> 5. Run in simulator... Another error!
> 
> > Failed to start launchd_sim: could not bind to session, launchd_sim may have crashed or quit responding Domain: com.apple.SimLaunchHostService.RequestError
> > [launchd_sim crashing: could not create temporary state directory](https://stackoverflow.com/questions/40238266/launchd-sim-crashing-could-not-create-temporary-state-directory)
> 
> 1. Run in simulator again... no error and worked!!
> 
> I hope this will be helpful for someone.

@tania628 Finally, this one worked for me. Thank you. :)

**Comment by lukaszciastko** [comment]
This issue is very confusing but pretty simple to fix. Unfortunately, the errors you're getting don't help much.

When you create a Flutter app, the target iOS version is set to 9.0. Most common plugins would have a minimum version higher than 9.0. If you add a plugin that has higher version, Flutter will complain. However, you can silence this error in two ways - either by setting a higher platform version in ios/Podfile or by using another plugin that has higher version number and depends on the initial plugin (e.g. multiple Firebase plugins). This will usually let you run the app in debug mode without issues, but when you try to build an API, the build will fail saying it cannot find a plugin (and it will be the first one in an alphabetical order, most likely device_info).

All you have to do is **set the higher iOS version in Xcode** (not in Podfile):

<img width="767" alt="Screenshot 2021-02-26 at 23 36 53" src="https://user-images.githubusercontent.com/964414/109362610-a80acf00-788b-11eb-9546-1c872f9a4e4a.png">

**Comment by ravindranathakila** [comment]
> Hi @rushabhshah065
> I came across the your comment [#43986 (comment)](https://github.com/flutter/flutter/issues/43986#issuecomment-605987016).
> Are you running from `Runner.xcworkspace`, right?

Really hope people upvote this more :)

**Comment by yaashwardhan** [comment]
> All of these together should solve it
> 
> 1. Open terminal, navigate to project then `flutter clean`
> 2. then navigate to /ios folder and `pods install`
> > Are you running from `Runner.xcworkspace`, right?
>
>_could give this a try too_ . Go to the project_file/ios folder launch the Runner.xcworkspace in Xcode, then click Runner > Targets > Runner > Build Settings, search Compile Sources As, change the value to Objective-C++

**Comment by douglasrosa0110** [comment]
you need re-create the ios folder.
1. Backup the current native folders (renaming the folders ./android and ./ios)
2. run in project root folder `flutter create .` (this will regenerate the native folders )
3. Your new Podfile now contains all you need.

`# Uncomment this line to define a global platform for your project
# platform :ios, '9.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
`

**Comment by KieranLafferty** [comment]
Resolved this myself by replacing all  `IPHONEOS_DEPLOYMENT_TARGET` values to match the minimum deployment target of my podfile (in my case 12.0)

**Comment by jmagman** [comment]
This issue has become pretty confused, and these symptoms could be caused by a few things already tracked in other issues.
1. Your app iOS version needs to match the minimum supported by your plugins.  The tool should make this much clearer and tell you how to fix the problem https://github.com/flutter/flutter/issues/56935
2. If you're launching from Xcode, always run from `ios/Runner.xcworkspace` https://github.com/flutter/flutter/issues/10770
3. You should get a better error now when you need to regenerate your Podfile.  https://github.com/flutter/flutter/pull/70735. We've tried recently to future-proof this a bit more, it should be pretty uncommon going forward to need to do this.  
4. If you have an Objective-C app and are trying to use a Swift app, you may need to add `use_frameworks!` to your `ios/Podfile` https://github.com/flutter/flutter/issues/16049#issuecomment-532476842

If you're seeing something similar and it's none of the above, please file a new issue and completely fill out the template.  Attach the output of `flutter run` with the `--verbose` flag, or the full Xcode build logs if you're seeing it from Xcode.

**Comment by yuhao900914** [comment]
In my case, I run the `pod install` first in the `ios` folder...  which is not needed. 
Should run `flutter build ios` for creating the Podfile.

**Comment by ChristianNavolskyi** [comment]
I also had the same problem. For me different schema names where the cause.
The schema name was `schema` but I named the build configuration `Schema`.
A dumb mistake, but maybe it helps someone.

**Comment by gj2569355373** [comment]
> > 嗨@rushabhshah065
> > 我看到了你的评论[#43986（评论）](https://github.com/flutter/flutter/issues/43986#issuecomment-605987016)。
> > 你在逃避`Runner.xcworkspace`，对吧？
> 
> 真的希望人们多点赞 :)

It works for me, thank you
